numpy/_core/src/multiarray/stringdtype/casts.cpp

```diff
@@ -1,13 +1,13 @@
-#include <cmath>
-#include <type_traits>
-
 #include "numpy/npy_common.h"
 #define PY_SSIZE_T_CLEAN
 #include <Python.h>
```

Comment:
Note that `npy_common.h` already includes  `Python.h`, so `PY_SSIZE_T_CLEAN` needs to be moved up. I think it would make sense to include it in `npy_common.h`.

> It is recommended to always define PY_SSIZE_T_CLEAN before including Python.h.

Reply:
> make sense to include it in npy_common.h

Although that might lead to redefinition problems. It is a problem for another day.

Reply:
From the error messages that led to this PR, I could see it going either way, so there are probably compilers on each side of that debate